exporter/datadogexporter/translate_traces.go

```diff
@@ -66,6 +66,8 @@ func convertToDatadogTd(td pdata.Traces, calculator *sublayerCalculator, cfg *co
 
 	var traces []*pb.TracePayload
 
+	spanMap := make(map[string]bool)
```

Comment:
Why do we only care about span IDs within a single batch? (This is intended given the PR description but I want to understand better). Also, should we set the capacity to `resourceSpans.Len()` since it will roughly be that?